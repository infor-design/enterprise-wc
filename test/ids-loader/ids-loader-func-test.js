/**
 * @jest-environment jsdom
 */
import IdsLoader from '../../src/ids-loader/ids-loader';

describe('IdsLoader Component', () => {
  let loader;

  beforeEach(async () => {
    const elem = new IdsLoader();
    document.body.appendChild(elem);
    loader = document.querySelector('ids-loader');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    document.body.innerHTML = '';
    loader = new IdsLoader();
    document.body.appendChild(loader);
    expect(document.querySelectorAll('ids-loader').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(loader.shadowRoot.innerHTML).toMatchSnapshot();
  });
});
