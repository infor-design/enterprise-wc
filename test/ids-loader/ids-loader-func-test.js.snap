// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsLoader Component renders correctly 1`] = `
"<style nonce=\\"0a59a005\\">.ids-loader { background-color: transparent; }</style><div class=\\"ids-loader\\">
      <div class=\\"ids-loader-indeterminate\\"></div>
    </div>"
`;
